cylc/flow/task_remote_mgr.py

```diff
@@ -165,44 +169,31 @@ def remote_init(self, platform, curve_auth,
                 A dictionary containing settings relating to platform used in
                 this remote installation.
 
-        Return:
-            REMOTE_INIT_NOT_REQUIRED:
-                If remote init is not required, e.g. not remote
-            REMOTE_INIT_DONE:
-                If remote init done.
-            REMOTE_INIT_FAILED:
-                If init of the remote failed.
-                Note: this will reset to None to allow retry.
-            None:
-                If waiting for remote init command to complete
-
         """
         install_target = platform['install target']
 
-        # If task is running locally or the install target is localhost
-        # we can skip the rest of this function
-        if (install_target == 'localhost' or
-                self.single_task_mode or
-                not is_remote_host(get_host_from_platform(platform))):
-            LOG.debug(f"REMOTE INIT NOT REQUIRED for {install_target}")
-            return REMOTE_INIT_NOT_REQUIRED
-
```

Comment:
Will need this logic somewhere? Otherwise could change the default in the remote-init map:

```
def __init__(..._):
    self.remote_init_map = {'localhost': REMOTE_INIT_DONE}
```

Reply:
I think in order to pickup the hosts defined as something other than localhost we will need to check is_remote_host at some stage. I've moved this out of remote_init so we do the check before going in here.